src/v2/Apps/Order/Routes/Review/index.tsx

```diff
@@ -122,13 +124,34 @@ export class ReviewRoute extends Component<ReviewProps> {
             }
           })
       } else {
-        this.props.order.conversation && !this.props.isEigen
-          ? this.props.router.push(
-              `/user/conversations/${this.props.order.conversation.internalID}`
-            )
-          : this.props.router.push(
-              `/orders/${this.props.order.internalID}/status`
-            )
+        const { order, router, user, isEigen } = this.props
+
+        if (!userHasLabFeature(user, "Make Offer On All Eligible Artworks")) {
+          return order.conversation && !isEigen
+            ? router.push(
+                `/user/conversations/${order.conversation.internalID}`
+              )
+            : router.push(`/orders/${order.internalID}/status`)
+        }
+        // Buy-mode order redirects to the status page
+        if (order.mode !== "OFFER") {
+          return router.push(`/orders/${order.internalID}/status`)
+        }
+        // Make offer in inquiry redirects to the conversation page
+        if (order.source === "inquiry" && !isEigen) {
+          return router.push(
+            `/user/conversations/${order.conversation?.internalID}`
+          )
+        }
+        // Make offer from artwork page redirects to the artwork page
+        const artworkId = get(
+          order,
+          o => o.lineItems?.edges?.[0]?.node?.artwork?.slug
+        )
+        return router.push({
+          pathname: `/artwork/${artworkId}`,
+          state: { offerOrderHasBeenSubmitted: true },
```

Comment:
This is a proposal to open the congrats modal window on the artwork page using the state on the router. To be challenged on [NX-3041](https://artsyproduct.atlassian.net/browse/NX-3041)